packages/dynamic-import/runtime/virtual-module.ts

```diff
@@ -14,6 +14,7 @@ import { srcDirName, lookupMap as _lookupMap } from "astro-dynamic-import:intern
 import type { SSRResult } from "astro"
 
 const lookupMap: Record<string, Promise<AstroComponentFactory>> = {}
+const processed: Array<string> = [];
```

Comment:
We would want to use a `WeakMap`, but other than that this seems great! Excellent work figuring this out!